src/cpp/session/modules/rmarkdown/SessionRMarkdown.cpp

```diff
@@ -1305,8 +1306,57 @@ Error maybeCopyWebsiteAsset(const json::JsonRpcRequest& request,
    return Success();
 }
 
-
-
+Error executeAlternateEngineChunk(const json::JsonRpcRequest& request,
+                                  json::JsonRpcResponse* pResponse)
+{
+   std::string docId, chunkId, engine, code;
+   Error error = json::readParams(request.params,
+                                  &docId,
+                                  &chunkId,
+                                  &engine,
+                                  &code);
+   if (error)
+   {
+      LOG_ERROR(error);
+      return error;
+   }
+   
+   if (string_utils::toLower(engine) == "python")
```

Comment:
I'm thinking that we shouldn't need to write engine specific code here as most engines should just take a script on the command line and hopefully won't care what the file extension is.